### PR TITLE
Result_storage with the option to store metadata

### PR DESCRIPTION
**Pull request by guilhermef**
This will add to aws, the ability to store metadata with the object.
It makes easy for any gateway to respond direct from S3.

### CODE REVIEW

**Comment by Bladrak** [comment]
This will store metadata in both result_storage & storage, is that what's needed? Don't we want them only in result storage?

File: tc_aws/aws/storage.py [hunk 1]
```diff
@@ -38,6 +38,10 @@ def set(self, bytes, abspath):
         file_key=Key(self.storage)
         file_key.key=abspath
 
+        if self.context.config.S3_STORE_METADATA:
```
Comment (Bladrak) [inline tc_aws/aws/storage.py]: Could you rather use the ``self.context.config.get(...)`` syntax? In order to harmonize with the rest of the code :)

**Comment by Bladrak** [comment]
Could you also add a mention within the ``README.md`` file about the new configuration key?

**Comment by guilhermef** [comment]
@Bladrak I've made a few changes, what do you think ?

**Comment by Bladrak** [comment]
Yep seems good to me, thanks for the contribution :)

**Pull request merged**